google/cloud/security/scanner/audit/org_rules_engine.py

```diff
@@ -269,22 +269,45 @@ def add_rule(self, rule_def, rule_index):
                                 bindings=rule_def.get('bindings', []),
                                 mode=rule_def.get('mode'))
 
+                    applies_to = resource.get('applies_to')
+
                     # See if we have a mapping of the resource and rule
-                    resource_rules = self.resource_rules_map.get(gcp_resource)
+                    resource_rules = self.resource_rules_map.get(
+                        (gcp_resource, applies_to))
 
                     # If no mapping exists, create it.
                     if not resource_rules:
                         resource_rules = ResourceRules(
                             resource=gcp_resource,
-                            applies_to=resource.get('applies_to'),
+                            applies_to=applies_to,
                             inherit_from_parents=rule_def.get(
                                 'inherit_from_parents', False))
-                        self.resource_rules_map[gcp_resource] = resource_rules
+                        self.resource_rules_map[
+                            (gcp_resource, applies_to)] = resource_rules
 
                     # If the rule isn't in the mapping, add it.
                     if rule not in resource_rules.rules:
                         resource_rules.rules.add(rule)
 
+    def _get_resource_rules(self, resource):
+        """Get all the resource rules for (resource, RuleAppliesTo.*).
+
+        Args:
+            resource: The resource to find in the ResourceRules map.
+
+        Returns:
+            A list of ResourceRules.
+        """
+        resource_rules = []
+        resource_rules.append(self.resource_rules_map.get(
+            (resource, RuleAppliesTo.SELF)))
+        resource_rules.append(self.resource_rules_map.get(
+            (resource, RuleAppliesTo.SELF_AND_CHILDREN)))
+        resource_rules.append(self.resource_rules_map.get(
+            (resource, RuleAppliesTo.CHILDREN)))
+
+        return [r for r in resource_rules if r]
```

Comment:
nit: if you decide not to do what andrew mention above, then the more pythonic way  to write this is:
```python
if r is not None
```
http://stackoverflow.com/a/3965129/2830207